### PR TITLE
Alewis/remove buckets

### LINKED ISSUE #634: Defer `kv:bucket` subcommands for later release
The functionality we've created around using kv as a bucket is tailored very specifically to the `site` use case. we can generalize this later, but for now we should remove the commands as they are not stable enough for release.

### PR DESCRIPTION
**Pull request by ashleymichal**
closes #634 . also removes `--no-worker` and `--no-bucket` flags from publish.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -705,57 +691,6 @@ fn run() -> Result<(), failure::Error> {
             ("", None) => message::warn("kv:bulk expects a subcommand"),
             _ => unreachable!(),
         }
-    } else if let Some(kv_matches) = matches.subcommand_matches("kv:bucket") {
```
Comment (gabbifish) [inline src/main.rs]: Need to remove all bucket logic in the Clap::App declaration above as well

Comment (ashleymichal) [reply src/main.rs]: did i not? lame.

Comment (gabbifish) [reply src/main.rs]: Just did for ya!

Comment (ashleymichal) [reply src/main.rs]: oh man you confused to 💩 out of me. thanks!

**Pull request merged**